### DISCUSSION TITLE
Empty module name at to_hls.InferStreamingMaxPool operation

### DISCUSSION THREAD

**Discussion by whitelambs** [Xilinx/finn#574] [Q&A]
Dear FINN team,

I have a custom CNV model with different dimension and input_size.
`import torch
from torch.nn import Module, ModuleList, BatchNorm2d, MaxPool2d, BatchNorm1d

from brevitas.nn import QuantConv2d, QuantIdentity, QuantLinear
from brevitas.core.restrict_val import RestrictValueType
from .tensor_norm import TensorNorm
from .common import CommonWeightQuant, CommonActQuant

CNV_OUT_CH_POOL = [(64, True), (128, True), (256, False), (256, True), (512, False), (512, True)]
INTERMEDIATE_FC_FEATURES = [(51200, 512), (512, 512)]
LAST_FC_IN_FEATURES = 512
LAST_FC_PER_OUT_CH_SCALING = False
POOL_SIZE = 2
KERNEL_SIZE = 3

class CNV(Module):

    def __init__(self, num_classes, weight_bit_width, act_bit_width, in_bit_width, in_ch):
        super(CNV, self).__init__()

        self.conv_features = ModuleList()
        self.linear_features = ModuleList()

        self.conv_features.append(QuantIdentity( # for Q1.7 input format
            act_quant=CommonActQuant,
            bit_width=in_bit_width,
            min_val=- 1.0,
            max_val=1.0 - 2.0 ** (-7),
            narrow_range=False,
            restrict_scaling_type=RestrictValueType.POWER_OF_TWO))

        for out_ch, is_pool_enabled in CNV_OUT_CH_POOL:
            self.conv_features.append(QuantConv2d(
                kernel_size=KERNEL_SIZE,
                in_channels=in_ch,
                out_channels=out_ch,
                bias=False,
                weight_quant=CommonWeightQuant,
                weight_bit_width=weight_bit_width))
            in_ch = out_ch
            self.conv_features.append(BatchNorm2d(in_ch, eps=1e-4))
            self.conv_features.append(QuantIdentity(
                act_quant=CommonActQuant,
                bit_width=act_bit_width))
            if is_pool_enabled:
                self.conv_features.append(MaxPool2d(kernel_size=2))

        for in_features, out_features in INTERMEDIATE_FC_FEATURES:
            self.linear_features.append(QuantLinear(
                in_features=in_features,
                out_features=out_features,
                bias=False,
                weight_quant=CommonWeightQuant,
                weight_bit_width=weight_bit_width))
            self.linear_features.append(BatchNorm1d(out_features, eps=1e-4))
            self.linear_features.append(QuantIdentity(
                act_quant=CommonActQuant,
                bit_width=act_bit_width))

        self.linear_features.append(QuantLinear(
            in_features=LAST_FC_IN_FEATURES,
            out_features=num_classes,
            bias=False,
            weight_quant=CommonWeightQuant,
            weight_bit_width=weight_bit_width))
        self.linear_features.append(TensorNorm())
        
        for m in self.modules():
          if isinstance(m, QuantConv2d) or isinstance(m, QuantLinear):
            torch.nn.init.uniform_(m.weight.data, -1, 1)

    def clip_weights(self, min_val, max_val):
        for mod in self.conv_features:
            if isinstance(mod, QuantConv2d):
                mod.weight.data.clamp_(min_val, max_val)
        for mod in self.linear_features:
            if isinstance(mod, QuantLinear):
                mod.weight.data.clamp_(min_val, max_val)

    def forward(self, x):
        x = 2.0 * x - torch.tensor([1.0], device=x.device)
        for mod in self.conv_features:
            x = mod(x)
        x = x.view(x.shape[0], -1)
        for mod in self.linear_features:
            x = mod(x)
        return x

def cnv(cfg):
    weight_bit_width = cfg.getint('QUANT', 'WEIGHT_BIT_WIDTH')
    act_bit_width = cfg.getint('QUANT', 'ACT_BIT_WIDTH')
    in_bit_width = cfg.getint('QUANT', 'IN_BIT_WIDTH')
    num_classes = cfg.getint('MODEL', 'NUM_CLASSES')
    in_channels = cfg.getint('MODEL', 'IN_CHANNELS')
    net = CNV(weight_bit_width=weight_bit_width,
              act_bit_width=act_bit_width,
              in_bit_width=in_bit_width,
              num_classes=num_classes,
              in_ch=in_channels)
    return net`

 I've trained my model mainly using https://github.com/Xilinx/brevitas/tree/master/src/brevitas_examples/bnn_pynq.

Then, I exported my model using FINNManager.export. During exportation, I received these warnings below.
![Screenshot from 2022-03-12 16-36-19](https://user-images.githubusercontent.com/40557991/158024549-d5578f08-52b6-4e34-ba5b-ac81420482c3.png)

After that, I've used same steps in end2end_example/bnn-pynq/cnv_end2end_example to build CNV using FINN. After streamline operation, my nodes was not correct like in the picture above.
![CNV_1W1A_streamlined onnx](https://user-images.githubusercontent.com/40557991/158024822-05881dfb-a847-4be5-b823-155a1d2060fd.png)
Therefore I've removed extra transpose layers and apply MakeMaxPoolNHWC again. As a result, I've received this onnx model.
![CNV_1W1A_fix onnx](https://user-images.githubusercontent.com/40557991/158024826-2a9b3f73-5d37-46bf-afcc-2d824204bb18.png)

I've started to apply Maxpool operation, but I'm constantly receiving this error.
![Screenshot from 2022-03-12 16-42-21](https://user-images.githubusercontent.com/40557991/158024732-761cd60d-4f6d-4f21-8c8d-26efc7f86369.png)

Thanks in advance for your help!
